pkg/actions/lua/databricks/client.go

```diff
@@ -0,0 +1,177 @@
+package databricks
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"regexp"
+	"strings"
+
+	"github.com/Shopify/go-lua"
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/config"
+	"github.com/databricks/databricks-sdk-go/service/catalog"
+	"github.com/databricks/databricks-sdk-go/service/sql"
+)
+
+// https://docs.databricks.com/en/sql/language-manual/sql-ref-identifiers.html
+var identifierRegex = regexp.MustCompile(`\W`)
+
+var (
+	ErrInvalidTableName     = errors.New("invalid table name")
+	ErrTableCreationFailure = errors.New("failed creating table")
+)
+
+type Client struct {
+	workspaceClient *databricks.WorkspaceClient
+	ctx             context.Context
+}
+
+// https://docs.databricks.com/en/sql/language-manual/sql-ref-identifiers.html
+// https://docs.databricks.com/en/sql/language-manual/sql-ref-names.html
+func validateTableName(tableName string) error {
+	if identifierRegex.Match([]byte(tableName)) {
+		return ErrInvalidTableName
+	} else if len(tableName) > 255 {
+		return ErrInvalidTableName
+	}
+	return nil
+}
+
+func validateTableLocation(tableLocation string) error {
+	_, err := url.ParseRequestURI(tableLocation)
+	return err
+}
+
+func validateTableInput(tableName, location string) error {
+	errName := validateTableName(tableName)
+	errLocation := validateTableLocation(location)
+	return errors.Join(errName, errLocation)
+}
+
+func (client *Client) createExternalTable(warehouseID, catalogName, schemaName, tableName, location string) (string, error) {
+	if err := validateTableInput(tableName, location); err != nil {
+		return "", errors.Join(ErrTableCreationFailure, err)
```

Comment:
Looking at the error composition and the one at line 65, where we fail to call `ExeciteAndWait - it seems that here you wanted to check that ErrTableCreationFailure and not the validation errors and in the execute you report the underlaying error.
So my question is do we need to report that the creation is failed without knowing the reason, or we need to know just that the creation failed.
If you need to understand that validation failed you mapped already an error so use fmt.Errorf("external table creation failed: %w", err) as you need on execute and wait.